pkg/scaling/scale_handler.go

```diff
@@ -432,6 +433,12 @@ func (h *scaleHandler) GetScaledObjectMetrics(ctx context.Context, scaledObjectN
 	isScalerError := false
 	scaledObjectIdentifier := scaledObject.GenerateIdentifier()
 
+	if _, present := scaledObject.GetAnnotations()[kedacontrollerutil.PausedReplicasAnnotation]; present {
+		ref := scaledObject.Spec.ScaleTargetRef
+		logger.Info("Detect ScaledObject with pause annotation", "targetKind", ref.Kind, "targetName", ref.Name)
+		prommetrics.RecordScalerPause(scaledObjectNamespace, scaledObjectName)
+	}
```

Comment:
I think that we need to evaluate this above, in the scaledobject_controller, as this is part of the scaledobject itself and not the scaling handler. WDYT @zroubalik ?
We need to decrease the counter too, atm once a scaledobject is paused and increased the counter, I guess that we never decrease it

Reply:
About that, I didn't know where is the best place to put this logic, so I decided to write it down and then discuss it here.

Reply:
yeah, we should also add Prometheus metric per ScaledObject - to mark whether is it paused or not - similar to `keda_scaler_activity` - https://keda.sh/docs/2.10/operate/prometheus/#operator

Reply:
I moved the logic to the controller as you ask. wdyt?

Reply:
I will change the metric from counter into gauge. As Prometheus docs said - "Do not use a counter to expose a value that can decrease. For example, do not use a counter for the number of currently running processes; instead use a gauge." It looks like the same case here. Will update the pr soon. Please let me know if there is something else that I'm missing